community/modules/database/slurm-cloudsql-federation/main.tf

```diff
@@ -33,9 +33,27 @@ locals {
   sql_password      = var.sql_password == null ? random_password.password.result : var.sql_password
 }
 
+resource "google_compute_global_address" "private_ip_address" {
```

Comment:
I would like to have a little more explanation here, are you changing this so it always use the private ip address?  I guess this makes sense for slurm, but don't we want to leave the option to also have a public address?

Reply:
I was thinking of this also Carlos. I couldn't come up with a valid use case where one would need this sql database to have a public IP. If needed one would have a direct peering or already another way of accessing the slurm controller node already. In the spirit of reducing public IP addresses and the attack surface, I would prefer if we would remove this as an option. If there is indeed a use case I am missing and it wouldn't be possible to be solved by a direct vpc peering then I agree with you we should have it as an option.

Reply:
My understanding is that the original use case was inter-regional database replication. Whether public IPs were actually required for that case, I can't say, but that's where I understood it to be coming from.

Reply:
That helps Tom thanks. Even the more reasons to actually force private vs public. Copying directly from our website I can see that for our replicas we also state that: "Creating a cross-region replica requires no networking setup. Global VPC uses private IP for replication traffic between regions—eliminating the need of complex VPN and VPC configuration, which would be otherwise needed to set up cross-region networking." and also: "Cross-region replication traffic remains private, without access to and from the public internet, when a private IP option is used." Based on these findings I would tend to agree with Eimantas and make private IP the default and only solution for now. If a use case comes up we can still revisit.

Reply:
I can state with certainty that this module is the only known reason `nat_ips` exists as an output for the `modules/network/vpc` module.

I also agree that it seems unlikely to me that public IPs were strictly necessary in the first place. I'm also not sure this feature is actively used via the Toolkit. From purely technical perspective, I agree with the conclusion here.